qcore/validate_vm.py

```diff
@@ -15,219 +15,238 @@
-                    bin_file, vm_size, size
-                ),
-            )
+    for file_path in vm_files:
+        # Test all files we can, so we get all the problems at once
+        if file_path.exists():
+            err = validate_vm_file(file_path, vm_size)
+            if err:
+                errors.extend(err)
+
+    # 8: Check VM within bounds -If DEM file is not present, fails the VM
 
-    # 6: binary contents
-    if numpy:
-        mins = []
-        for file_name in vm.values():
-            mins.append(
-                np.min(
-                    np.fromfile(
-                        file_name,
-                        dtype="<f{}".format(SIZE_FLOAT),
-                        count=vm_params_dict[VMParams.nz.value]
-                        * vm_params_dict[VMParams.nx.value],
+    polygon = []
+
+    if vel_crns_file.exists():
+        with open(vel_crns_file) as crns_fp:
+            next(crns_fp)
+            next(crns_fp)
+            for line in crns_fp:
+                lon, lat = map(float, line.split())
+                lon = lon % 360
+                polygon.append((lon, lat))
+                if lon < MIN_LON or lon > MAX_LON or lat < MIN_LAT or lat > MAX_LAT:
+                    errors.append("VM extents not contained within NZVM DEM")
+
+        # 9: Check SRF is within bounds of the VM if it is given
+        if srf is not None:
+            srf_bounds = get_bounds(srf)
+            edges = []
+            for index, start_point in enumerate(polygon):
+                end_point = polygon[(index + 1) % len(polygon)]
+                lons = (
+                    np.linspace(
+                        start_point[0],
+                        end_point[0],
+                        int(ll_dist(*start_point, *end_point)),
                     )
+                    % 360
                 )
-            )
+                lats = compute_intermediate_latitudes(start_point, end_point, lons)
+                edges.extend(list(zip(lons, lats)))
+            path = mpltPath.Path(edges)
+            for bounds in srf_bounds:
+                if not all(path.contains_points(bounds)):
+                    errors.append(
+                        "Srf extents not contained within velocity model corners"
+                    )
+    if errors:
+        return False, "\n".join(errors)
+    return True, ""
 
-        # works even if min is np.nan
-        if not min(mins) > 0:
-            return False, "VM vs, vp or rho <= 0|nan found: {}".format(vm_dir)
 
-    # 7: contents of meta files
-    #    if meta_created:
-    #        # TODO: check individual file contents
-    #        # not as important, can be re-created based on vm_params.py
-    #        pass
+def validate_vm_file(file_name, vm_size):
+    errors = []
+    size = file_name.stat().st_size
+    if size != vm_size * SIZE_FLOAT:
+        errors.append(
+            f"VM filesize for {file_name} expected: {vm_size * SIZE_FLOAT} found: {size}"
+        )
+    if (
```

Comment:
I'd add a comment here. Actually a docstring might be clearest. Would be good to state what is being tested here.